File: backend/src/main/java/br/ucs/greencitizenship/entities/enums/StatusEnum.java
package br.ucs.greencitizenship.entities.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum StatusEnum {

    IN_REVISION(1, "In Revision"),
    IN_PROGRESS(2, "In Progress"),
    COMPLETED(3, "Completed"),
    CANCELED(4, "Canceled");

    private final Integer value;
    private final String label;

    public static StatusEnum fromValue(Integer value) {
        for (StatusEnum obj : StatusEnum.values()) {
            if (obj.value.equals(value)) {
                return obj;
            }
        }
        throw new IllegalArgumentException("Unknown value: " + value);
    }
}

File: frontend/src/models/enums/statusEnum.ts
export const DStatusEnum = {
    IN_REVISION: { name: "IN_REVISION", value: 1, label: "Em revisão" },
    IN_PROGRESS: { name: "IN_PROGRESS", value: 2, label: "Em andamento" },
    COMPLETED: { name: "COMPLETED", value: 3, label: "Concluído" },
    CANCELED: { name: "CANCELED", value: 4, label: "Cancelado" }
} as const;